utils/common/src/main/kotlin/Extensions.kt

```diff
@@ -438,3 +438,35 @@ fun URI.getQueryParameters(): Map<String, List<String>> {
         .groupBy({ it.substringBefore('=') }, { it.substringAfter('=').split(',') })
         .mapValues { (_, v) -> v.flatten() }
 }
+
+/**
+ * Return a string of common-separated ranges as denoted by the list of pairs.
+ */
+fun List<Pair<Int, Int>>.prettyPrintRanges(): String {
```

Comment:
The functions in this file should be sorted alphabetically, i.e. by the name of the class they extend. To avoid these functions being separated due to sorting, I'd propose to make this extend a `Collection` instead of a `List` as well (even if not meaningful for our current use-case).